projects/graph_class/graphs.js

```diff
@@ -11,27 +11,78 @@ class Graph {
      * }
      *
      */
-    constructor() {}
+    constructor() {
+      this.graph = {}
+    }
+
   
     //adds a node to the graph
-    addVertex() {}
+    addVertex(vertex) {
+      this.graph[vertex] = []
+    }
     
     //adds an edge to the graph
-    addEdge() {}
+    addEdge(vertex, edge) {
+      for(let vert in this.graph){
+        if(vert === vertex){
+          this.graph[vertex].push(edge)
+        }
+      }
+    }
   
     //returns the neighbors (edges) for a particular vertex
-    getNeighbors() {}
+    getNeighbors(vertex) {
+      for(let vert in this.graph){
+        if(vert === vertex){
+          return this.graph[vertex]
+        }
+      }
+    }
   
-    breadthFirstTraversal() {
-        let visited = []
-        
-        this.printPath(visited)
+    breadthFirstTraversal(startVert) {
+      let visited = []
+      let queue = [startVert]
+
+      while(queue.length > 0){
+        const vert = queue.shift()
+        if(vert && !queue.includes(vert)){
+          if(!visited.includes(vert)){
+              visited.push(vert)
+
+              let neighbors = this.getNeighbors(vert)
+
+              neighbors.forEach(neighbor => {
+                if(!visited.includes(neighbor)){
+                  queue.push(neighbor)
+                }
+              })
+          }
+        }
+      }
+      this.printPath(visited)
     }
   
-    depthFirstTraversal() {
-        let visited = []
-        
-        this.printPath(visited)
+    depthFirstTraversal(startVert) {
+      let visited = []
+      let queue = [startVert]
```

Comment:
nit: in Depth first it's a stack rather than a queue. While they're both arrays since you're popping from the array it's behaving more as a stack LIFO rather than a queue FIFO